File: problems/2327_number_of_people_aware_of_a_secret/2327_number_of_people_aware_of_a_secret.py
class Solution:
    def peopleAwareOfSecret(self, n: int, delay: int, forget: int) -> int:
        MOD = 10**9 + 7
        dp = [0] * n
        dp[0] = 1
        share = 0
        for i in range(1, n):
            if i >= delay:
                share = (share + dp[i - delay] + MOD) % MOD
            if i >= forget:
                share = (share - dp[i - forget] + MOD) % MOD
            dp[i] = share
        return sum (dp[n - forget:]) % MOD

File: problems/2327_number_of_people_aware_of_a_secret/2327_number_of_people_aware_of_a_secret.java
class Solution {
    public int peopleAwareOfSecret(int n, int delay, int forget) {
        final int MOD = 1_000_000_007;
        long share = 0;
        int[] dp = new int[n];
        dp[0] = 1;
        for (int i = 1; i < n; i++) {
            if (i >= delay) {
                share = (share + dp[i - delay] + MOD) % MOD;
            }
            if (i >= forget) {
                share = (share - dp[i - forget] + MOD) % MOD;
            }
            dp[i] = (int) share;
        }
        long result = 0;
        for (int i = n - forget; i < n; i++) {
            result = (result + dp[i]) % MOD;
        }
        return (int) result;
    }
}